homeassistant/components/islamic_prayer_times/coordinator.py

```diff
@@ -0,0 +1,122 @@
+"""Coordinator for the Islamic prayer times integration."""
+from __future__ import annotations
+
+from datetime import datetime, timedelta
+import logging
+
+from prayer_times_calculator import PrayerTimesCalculator, exceptions
+from requests.exceptions import ConnectionError as ConnError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import CALLBACK_TYPE, HomeAssistant
+from homeassistant.helpers.event import async_call_later, async_track_point_in_time
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+import homeassistant.util.dt as dt_util
+
+from .const import CONF_CALC_METHOD, DEFAULT_CALC_METHOD, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class IslamicPrayerDataUpdateCoordinator(DataUpdateCoordinator[dict[str, datetime]]):
+    """Islamic Prayer Client Object."""
+
+    config_entry: ConfigEntry
+
+    def __init__(self, hass: HomeAssistant, config_entry: ConfigEntry) -> None:
+        """Initialize the Islamic Prayer client."""
+        self.hass = hass
+        self.config_entry = config_entry
+        self.event_unsub: CALLBACK_TYPE | None = None
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=DOMAIN,
+        )
+
+    @property
+    def calc_method(self):
+        """Return the calculation method."""
+        return self.config_entry.options.get(CONF_CALC_METHOD, DEFAULT_CALC_METHOD)
+
+    def get_new_prayer_times(self) -> dict[str, str]:
+        """Fetch prayer times for today."""
+        calc = PrayerTimesCalculator(
+            latitude=self.hass.config.latitude,
+            longitude=self.hass.config.longitude,
+            calculation_method=self.calc_method,
+            date=str(dt_util.now().date()),
+        )
+        return calc.fetch_prayer_times()
+
+    async def async_schedule_future_update(self, midnight_dt: datetime) -> None:
```

Comment:
This doesn't need to be a coroutine function as there's no await inside.

```suggestion
    @callback
    def async_schedule_future_update(self, midnight_dt: datetime) -> None:
```